reconcile/utils/prometheus.py

```diff
@@ -61,13 +61,14 @@ class PrometheusHttpQuerier(BaseModel):
     auth_token: str
 
     def instant_vector_query(self, query: str) -> list[PrometheusVector]:
-        response = requests.get(
-            self.query_url,
-            params={"query": query},
-            headers={
-                "Authorization": f"Bearer {self.auth_token}",
-            },
-        )
+        with requests.Session() as session:
+            response = session.get(
+                self.query_url,
+                params={"query": query},
+                headers={
+                    "Authorization": f"Bearer {self.auth_token}",
+                },
+            )
```

Comment:
`requests.get` is using `with sessions.Session() as session` internally, so this won't cause memory leak.

But if we want to reuse the same connection, we should manage session in a field or pass in as an argument, then if this request called multiple times in one run, we should see same fd reuse instead keep creating and close, that's a performance improvement task.

Reply:
reuse is and not worth the effort here. we talk about one request per reconcile run and OCM environment, so three for the time being.

i reverted this particular change